File: services/rails-modular/db/migrate/20260828000001_create_users.rb
# frozen_string_literal: true

class CreateUsers < ActiveRecord::Migration[8.1]
  def change
    # Cognito の sub をそのまま主キーにするため、自動採番の id を使わない。
    create_table :users, id: false do |t|
      t.string :id, null: false, primary_key: true, limit: 64
      t.string :email, null: false
      t.string :display_name, null: false, limit: 50
      t.boolean :active, null: false, default: true

      t.timestamps
    end

    add_index :users, :email, unique: true
  end
end

File: services/cakephp-modular/config/Migrations/20260828000001_CreateUsers.php
<?php
declare(strict_types=1);

use Migrations\BaseMigration;

/**
 * users テーブル。
 *
 * Cognito の sub をそのまま主キーにするため、自動採番の id を使わない。
 */
class CreateUsers extends BaseMigration
{
    public function change(): void
    {
        $this->table('users', ['id' => false, 'primary_key' => ['id']])
            ->addColumn('id', 'string', ['limit' => 64, 'null' => false])
            ->addColumn('email', 'string', ['limit' => 254, 'null' => false])
            ->addColumn('display_name', 'string', ['limit' => 50, 'null' => false])
            ->addColumn('is_active', 'boolean', ['null' => false, 'default' => true])
            ->addColumn('created', 'datetime', ['null' => false, 'default' => 'CURRENT_TIMESTAMP'])
            ->addColumn('modified', 'datetime', ['null' => false, 'default' => 'CURRENT_TIMESTAMP'])
            ->addIndex(['email'], ['unique' => true])
            ->create();
    }
}
